### PR TITLE
Release 3.3.1

### PR DESCRIPTION
**Pull request by FinalAngel**
- [x] Finalize test documentation
- [x] Implement proper tests for base setup

Bonus:

- [ ] Setup SauceLabs

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -134,6 +135,13 @@ gulp.task('tests', ['lint'], function () {
     });
 });
 
+gulp.task('karma', function () {
```
Comment (vxsx) [inline gulpfile.js]: maybe make it part of gulp watch?

File: tests/test.cl.utils.js [hunk 2]
```diff
@@ -28,6 +28,47 @@
...
+        describe('.setStorage() and .getStorage()', function () {
+            beforeEach(function () {
+                localStorage.clear();
+            });
+
+            it('use the local storage of the browser', function () {
+                if (typeof (Storage) !== void(0)) {
+                    expect(Storage).toEqual(jasmine.any(Object));
+                } else {
+                    expect(Storage).toThrowError(TypeError);
```
Comment (vxsx) [inline tests/test.cl.utils.js]: isn't it going to be `ReferenceError`?

**Comment by vxsx** [comment]
I expressed my concerns regarding storage checks [here](https://github.com/aldryn/aldryn-boilerplate-bootstrap3/issues/70) but that can be addressed later.

Other than that, great job.

File: gulpfile.js (hunk 1 above)
Comment (FinalAngel) [inline gulpfile.js]: I'm not so sure yet, want to play around a bit more and involve @Marketionist with the testing setup before doing so. Sounds alright?

File: tests/test.cl.utils.js (hunk 2 above)
Comment (FinalAngel) [inline tests/test.cl.utils.js]: correct

File: gulpfile.js (hunk 1 above)
Comment (vxsx) [inline gulpfile.js]: yeah sure

**Comment by vxsx** [comment]
LGTM

**Pull request merged**